src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Lock/ResourceLockManagementCmdletBase.cs

```diff
@@ -91,6 +91,7 @@ public abstract class ResourceLockManagementCmdletBase : ResourceManagerCmdletBa
         /// </summary>
         [Parameter(ParameterSetName = ResourceLockManagementCmdletBase.ResourceGroupResourceLevelLock, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name.")]
         [Parameter(ParameterSetName = ResourceLockManagementCmdletBase.ResourceGroupLevelLock, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name.")]
+        [Parameter(ParameterSetName = ResourceLockManagementCmdletBase.SubscriptionResourceLevelLock, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name.")]
         [ValidateNotNullOrEmpty]
```

Comment:
@gandhiniraj  ResourceGroupName should not be a part of the SubscriptionResourceLevelLock parameter set.  The SubscriptionResourceLevelLock is for locks at the subscription level, like /subscriptions/{subId}/RPNamespace/RPType/lockname.

Is there an issue opened for the New-AzureRmResourceLock cmdlet? I just tried creating a lock at using the -Scope and it works. If there is an issue with the default parameter set, we should instead investigate that. Currently, I see the subscription level lock being set as the default for the new cmdlet. We should instead use the ScopeLevel parameter set as the default one. Agree @ravbhatnagar ?